src/utils/chart.ts

```diff
@@ -0,0 +1,120 @@
+import { graphic } from "echarts/core";
+
+function formatter(value: Number) {
+  return value >= 100000 ? value.toExponential() : value;
+}
+
+function getSeries(dataType: string, chartType: string, color: string, data: Array<any>) {
+  return {
+    name: dataType,
+    data: data,
+    type: chartType,
+    itemStyle: {
+      color: color,
+    },
+    lineStyle: {
+      color: color,
+    },
+    areaStyle: {
+      color: new graphic.LinearGradient(0, 0, 0, 1, [
+        {
+          offset: 0,
+          color: color + "a0", // hex rgba
+        },
+        {
+          offset: 1,
+          color: color + "09",
+        },
+      ]),
+    },
+  };
+}
+
+function minimumChart(timeSeries: object, chartType: string, dataType: string, color: string) {
```

Comment:
I think TypeScript can allow one to specify the structural typing, to be more specific than just `object`.